app/classifier/tasks/survey/choice.spec.js

```diff
@@ -31,7 +31,7 @@ describe('Choice', function () {
 
   describe('with single answer questions', function () {
     beforeEach(function () {
-      wrapper = shallow(<Choice
+      wrapper = mount(<Choice
```

Comment:
`mount` seems to be working but it's a long time since I've worked with the survey task and I couldn't remember what a shallow-rendered choice looks like. The only cases I've found where I've had to use `mount` in tests are components that use refs, which aren't supported by shallow rendering.

Reply:
I've found that you have to use `mount` when a function has to be called in the child renders, so in this case, the actual label wasn't rendering at the start with shallow because I think the counterpart function hadn't run to render the label. The choice is either use the dive utility or use mount.

Reply:
The survey task code is pretty old. Those functions should be testable components in their own right. Or maybe just tested independently here.

Reply:
No wait, I think `mount` is being used here to fully render a `TriggeredModalForm` component? That's the only place where I can see the label being used. Those labels aren't rendered with counterpart, they're supplied from the API.

Reply:
I changed it back to shallow.

I'm not testing [the content in the modal when you click a confused with option](https://github.com/zooniverse/Panoptes-Front-End/pull/5540/files#diff-3bb4a3465c95d0e6d761e7d94f49f939R137), maybe that's what you're thinking of? I think that might require `mount`, or more than I think is necessary for now?

Reply:
Looking at the code, I think you're testing the `trigger` prop on the modal form. That's the only place where I can see the label being used. Fully rendering the DOM will render each trigger as a styled `<span>` in the Choice component.

Reply:
That's correct, testing the `trigger` prop on the modal form, and I think `mount` or `dive` gets the desired result. I think using `shallow` throughout the spec is a good idea, so currently using that and `dive`, which I'm open to change, but if that's ok I think the rest of this is good to go as well.

Reply:
I was thinking `shallow` plus expecting the `trigger` prop for each modal form to have a particular value but this works.